scripts/jenkins-yoctobuild-build.sh

```diff
@@ -130,35 +147,15 @@ EOF
                 # Not a pull request, skip.
                 continue
             fi
-            case "$key" in
-                META_MENDER_REV)
-                    local repo=meta-mender
-                    local location=$WORKSPACE/meta-mender
-                    ;;
-                *_REV)
-                    local repo=$(tr '[A-Z_]' '[a-z-]' <<<${key%_REV})
-                    if ! $WORKSPACE/integration/extra/release_tool.py --version-of $repo; then
-                        # If the release tool doesn't recognize the repository, don't use it.
-                        continue
-                    fi
-                    local location=
-                    if [ -d "$WORKSPACE/$repo" ]; then
-                        location="$WORKSPACE/$repo"
-                    elif [ -d "$WORKSPACE/go/src/github.com/mendersoftware/$repo" ]; then
-                        location="$WORKSPACE/go/src/github.com/mendersoftware/$repo"
-                    else
-                        echo "github_pull_request_status: Unable to find repository location: $repo"
-                        return 1
-                    fi
-                    ;;
-                *)
-                    # Not a revision, go to next entry.
-                    continue
-                    ;;
-            esac
-            local git_commit=$(cd "$location" && git rev-parse HEAD)
-            local pr_status_endpoint=https://api.github.com/repos/mendersoftware/$repo/statuses/$git_commit
 
+            local repo=$(tr '[A-Z_]' '[a-z-]' <<<${key%_REV})
+            local pullnr=$(eval echo \$$key | cut -d/ -f2)
+            local pr_info_endpoint="https://api.github.com/repos/mendersoftware/$repo/pulls/$pullnr"
+
+            # First API call to get status endpoint https://developer.github.com/v3/pulls/#get-a-single-pull-request
+            local pr_status_endpoint=$(curl --user "$GITHUB_BOT_USER:$GITHUB_BOT_PASSWORD" $pr_info_endpoint | jq -r '.statuses_url')
```

Comment:
I think there is a race condition here, because the PR might have been updated after you first pulled it, and then the wrong commit may be marked as passing. I would expect this to be a fairly common occurrence when someone is first submitting a PR and making small adjustments. I think querying the local commit is a safer method.

Maybe to find the repo more easily, you can do something like:

```
location="$(find $WORKSPACE -type d -name "$repo")"
```

Reply:
Good point.

The reason behind it (explained in the commit) is that the integration tests jobs do not have local clones of the repos, only `mender-qa` and `integration`.

Any other good idea to do this? Else I will need to clone all the repos for these jobs only for this.

Reply:
I reworked my proposal. Please see commit 5235662